packages/node_modules/@webex/plugin-meetings/test/unit/spec/meeting/index.js

```diff
@@ -0,0 +1,715 @@
+          it('should return a promise resolution', async () => {
+            const toggle = sinon.stub().returns(Promise.resolve(true));
+            meeting.audio = {toggle};
+            const audio = meeting.unmuteAudio();
+            assert.exists(audio.then);
+            await audio;
+            assert.calledOnce(toggle);
+            assert.calledWith(toggle, {mute: false, self: true});
+          });
+        });
+      });
+      describe('#muteVideo', () => {
+        describe('before video is defined', () => {
+          it('should reject and return a promise', async () => {
+            await meeting.muteVideo().catch((err) => {
+              assert.instanceOf(err, Error);
+            });
+          });
+        });
+        describe('after video is defined', () => {
+          it('should return a promise resolution', async () => {
+            const toggle = sinon.stub().returns(Promise.resolve(true));
+            meeting.video = {toggle};
+            const video = meeting.muteVideo();
+            assert.exists(video.then);
+            await video;
+            assert.calledOnce(toggle);
+            assert.calledWith(toggle, {mute: true, self: true});
+          });
+        });
+      });
+      describe('#unmuteVideo', () => {
+        describe('before video is defined', () => {
+          it('should reject and return a promise', async () => {
+            await meeting.unmuteVideo().catch((err) => {
+              assert.instanceOf(err, Error);
+            });
+          });
+        });
+        describe('after video is defined', () => {
+          it('should return a promise resolution', async () => {
+            const toggle = sinon.stub().returns(Promise.resolve(true));
+            meeting.video = {toggle};
+            const video = meeting.unmuteVideo();
+            assert.exists(video.then);
+            await video;
+            assert.calledOnce(toggle);
+            assert.calledWith(toggle, {mute: false, self: true});
+          });
+        });
+      });
+      describe('#joinWithMedia', () => {
+        describe('resolution', () => {
+          it('should success and return a promise', async () => {
+            meeting.join = sinon.stub().returns(Promise.resolve('test'));
+            meeting.getMediaStreams = sinon.stub().returns(Promise.resolve(['test', 'test']));
+            meeting.addMedia = sinon.stub().returns(Promise.resolve('test'));
+            await meeting.joinWithMedia();
+            assert.calledOnce(meeting.join);
+            assert.calledOnce(meeting.getMediaStreams);
+          });
+        });
+        describe('rejection', () => {
+          it('should error out and return a promise', async () => {
+            meeting.join = sinon.stub().returns(Promise.reject());
+            meeting.getMediaStreams = sinon.stub().returns(true);
+            await meeting.joinWithMedia();
+            assert.calledOnce(meeting.join);
+            assert.notCalled(meeting.getMediaStreams);
```

Comment:
doesn't joinWithMedia call getMediaStreams internally? I don't know why we would assert that we don't (or why it passes)

Reply:
likely commenting on a resolution vs rejection describe block

```
        describe('resolution', () => {
          it('should success and return a promise', async () => {
            meeting.join = sinon.stub().returns(Promise.resolve('test'));
            meeting.getMediaStreams = sinon.stub().returns(Promise.resolve(['test', 'test']));
            meeting.addMedia = sinon.stub().returns(Promise.resolve('test'));
            await meeting.joinWithMedia();
            assert.calledOnce(meeting.join);
            assert.calledOnce(meeting.getMediaStreams);
          });
        });
        describe('rejection', () => {
          it('should error out and return a promise', async () => {
            meeting.join = sinon.stub().returns(Promise.reject());
            meeting.getMediaStreams = sinon.stub().returns(true);
            await meeting.joinWithMedia();
            assert.calledOnce(meeting.join);
            assert.notCalled(meeting.getMediaStreams);
          });
        });
```